spec/token/ERC20/ERC20Base.behavior.ts

```diff
@@ -0,0 +1,177 @@
+import { expect } from 'chai';
+import { describeFilter } from '@solidstate/library/mocha_describe_filter';
+import { ethers } from 'hardhat';
+import { ERC20Base } from '../../../typechain';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+interface ERC20BaseBehaviorArgs {
+  deploy: () => Promise<ERC20Base>;
+  supply: ethers.BigNumber;
+  mint: (address: string, amount: ethers.BigNumber) => Promise<void>;
+  burn: (address: string, amount: ethers.BigNumber) => Promise<void>;
+}
+
+export function describeBehaviorOfERC20Base(
+  { deploy, supply, mint, burn }: ERC20BaseBehaviorArgs,
+  skips: string[],
+) {
+  const describe = describeFilter(skips);
+
+  describe('::ERC20Base', function () {
+    // note: holder gets supply (1e18) amount of tokens so use spender/receiver for easier testing
+    let holder: SignerWithAddress;
+    let spender: SignerWithAddress;
+    let receiver: SignerWithAddress;
+    let sender: SignerWithAddress;
+    let instance: ERC20Base;
+
+    before(async function () {
+      [holder, spender, receiver, sender] = await ethers.getSigners();
+    });
+
+    beforeEach(async function () {
+      instance = await deploy();
```

Comment:
How does Typescript handle casting of contracts?  If a third party contract has both `balanceOf(address)` and a `balanceOf(address,uint256)` functions, for example, will the calls in this file fail due to the overload?

Reply:
Good point.
So by default, types with method signatures are only generated when there is an overload, but typechain has a `--always-generate-overloads` options, which generate types with method signature for everything.

So I'll update to use this flag, and will use everywhere the function call through the method signature, which will solve this.